.travis.yml

```diff
@@ -4,27 +4,48 @@ env:
   - PYTHON_VERSION=$TRAVIS_PYTHON_VERSION
   - NUMPY_VERSION=stable
   - MAIN_CMD='python setup.py'
-  - CONDA_DEPENDENCIES='netcdf4 hdf5 coveralls coverage pytest pytest-cov'
+  - CONDA_DEPENDENCIES='netcdf4 hdf5 udunits2 coveralls coverage pytest pytest-cov'
   - PIP_DEPENDENCIES=''
-  - SETUP_XVFB=False
+  - SETUP_XVFB=True
   - EVENT_TYPE='push pull_request'
   - SETUP_CMD='test'
   - CONDA_CHANNELS='conda-forge'
   - CONDA_CHANNEL_PRIORITY="strict"
-matrix:
+  - UNSTABLE_DEPS=False
+  - FOO_BAR=cron  # TODO: replace with TRAVIS_EVENT_TYPE
+jobs:
   include:
-  - env: PYTHON_VERSION=3.6
+  - name: "Python 3.7"
+    env: PYTHON_VERSION=3.7
     os: linux
-  - env: PYTHON_VERSION=3.7
+  - name: "Python 3.7 (unstable)"
+    env:
+      - PYTHON_VERSION=3.7
+      - UNSTABLE_DEPS=True
     os: linux
-  - env: PYTHON_VERSION=3.8
+  - name: "Python 3.8"
+    env: PYTHON_VERSION=3.8
     os: linux
+
+  allow_failures:
+  - name: "Python 3.7 (unstable)"
+
 install:
 - git clone --depth 1 git://github.com/astropy/ci-helpers.git
 - source ci-helpers/travis/setup_conda.sh
-- pip install -e .
+- if [[ "$UNSTABLE_DEPS" == "True" ]]; then
+    pip install
+    git+https://github.com/pydata/xarray
+    git+https://github.com/pydata/pandas
+    git+https://github.com/pytroll/satpy
+    git+https://github.com/pytroll/pygac;
+  fi
+- pip install -e .[tests]
+before_script:
+- if [[ $PYTHON_VERSION == 3.7 && $FOO_BAR == "cron" ]]; then python pygac_fdr/tests/fetch_data.py; fi
 script:
-- pytest --cov=pygac_fdr pygac_fdr/tests
+- pytest --cov=pygac_fdr pygac_fdr/tests --ignore=pygac_fdr/tests/test_end2end.py
+- if [[ $PYTHON_VERSION == 3.7 && $FOO_BAR == "cron" ]]; then python pygac_fdr/tests/test_end2end.py; fi
```

Comment:
Should it be foo bar here?

Reply:
I'll change this to `TRAVIS_EVENT_TYPE` (== 'cron' if triggered by a cron job) as soon as I'm done with testing. For some reason Travis doesn't complete the build if I set that variable (or even only `EVENT_TYPE`) manually.

Reply:
ok